source/common/res/features/spare-change/main.js

```diff
@@ -0,0 +1,187 @@
+            .attr('title', 'The selected items "spare change" when rounded up to the nearest dollar.')
+            .text('Spare Change');
+
+        var currencySpan = $('<span />').addClass('user-data currency');
+
+        spareChangeAmount.append(currencySpan);
+        spareChangeDiv.append(spareChangeTitle);
+        spareChangeDiv.append(spareChangeAmount);
+
+        if (isSelectedTotalVisible) {
+          spareChangeDiv.insertBefore(selectedTotalDiv);
+          separator = $('<i />').text('***');
+          separator.insertAfter(spareChangeDiv);
+        } else {
+          spareChangeDiv.css('margin-right', '32px');
+          balancesHeader.append(spareChangeDiv);
+        }
+
+        flexGrowSpacer = $('<div />')
+          .addClass('ynab-toolkit-spare-change-flex-grow-spacer')
+          .css('flex-grow', '1');
+        flexGrowSpacer.insertBefore(spareChangeDiv);
+      }
+
+      function removeHeader() {
+        $('.ynab-toolkit-accounts-header-balances-spare-change').remove();
+        $('.accounts-header-balances > i:contains("***")').remove();
+        $('.ynab-toolkit-spare-change-flex-grow-spacer').remove();
+      }
+
+      function updateValue() {
+        var selectedAccount = getSelectedAccount();
+        if (selectedAccount) {
+          var spareChange = selectedAccount.__ynabToolKitSpareChange;
+
+          var spareChangeHeader = $('.ynab-toolkit-accounts-header-balances-spare-change');
+          var spareChangeAmount = $('.user-data:not(.currency)', spareChangeHeader);
+          var currencySpan = $('.user-data.currency', spareChangeHeader);
+
+          currencySpan.removeClass('negative positive zero');
+
+          if (spareChange < 0) {
+            currencySpan.addClass('negative');
+          } else if (spareChange > 0) {
+            currencySpan.addClass('positive');
+          } else {
+            currencySpan.addClass('zero');
+          }
+
+          var formatted = ynabToolKit.shared.formatCurrency(spareChange);
+          spareChangeAmount.attr('title', formatted.string);
+
+          var formattedHtml = formatted.string.replace(/\$/g, '<bdi>$</bdi>');
+          currencySpan.html(formattedHtml);
+        }
+      }
+
+      function onYnabGridyBodyChanged() {
+        Ember.run.later(function () {
+          setSelectedTransactions();
+          updateSpareChangeCalculation();
+          updateSpareChangeHeader();
+        }, 250);
+      }
+
+      function onYnabSelectionChanged() {
+        selectedTransactions = undefined;
+        onYnabGridyBodyChanged();
+      }
+
+      return {
+        // invoke has potential of being pretty processing heavy (needing to sort content, then add calculation to every row)
+        // wrapping it in a later means that if the user continuously scrolls down we won't clog up the event loop.
+        invoke: function invoke() {
+          currentlyRunning = true;
+
+          Ember.run.later(function () {
```

Comment:
It might be a good idea to use `Ember.run.debounce` instead of `Ember.run.later` so that it doesn't end up re-calculating 50 times if you scroll 50 times in rapid succession. `Ember.run.later` is definitely better than nothing, but debounce sounds like it's what you're trying to do to me.